lib/msal-common/src/request/RequestParameterBuilder.ts

```diff
@@ -608,6 +612,35 @@ export class RequestParameterBuilder {
             queryParameterArray.push(`${key}=${value}`);
         });
 
+        const correlationId = this.parameters.get(
+            AADServerParamKeys.CLIENT_REQUEST_ID
+        );
+        const clientId = this.parameters.get(AADServerParamKeys.CLIENT_ID);
+
+        if (
+            this.parameters.has(AADServerParamKeys.BROKER_CLIENT_ID) &&
+            clientId &&
+            correlationId
+        ) {
+            this.performanceClient?.addFields(
+                {
+                    embeddedClientId: clientId,
```

Comment:
I wonder, organizationally, if it makes sense to do this below where we add the embedded clientId to the params. I'm thinking this might be a pattern we want to adopt for other params as well in the future and it might help to keep things organized. 

If you agree, I'd also suggest passing correlationId into the constructor above and making it a member variable to better facilitate this.

Reply:
Added `correlationId` to the constructor to facilitate future telemetry instrumentation. However, broker param instrumentation is still invoked from `createQueryString` as we need embedded params to be picked up from the param map.